zio/detector/reader.go

```diff
@@ -11,50 +11,68 @@ import (
 	"github.com/brimsec/zq/zio/zjsonio"
 	"github.com/brimsec/zq/zio/zngio"
 	"github.com/brimsec/zq/zng/resolver"
+	"github.com/brimsec/zq/zqe"
 )
 
-var ErrUnknown = errors.New("malformed input")
-
 func NewReader(r io.Reader, zctx *resolver.Context) (zbuf.Reader, error) {
 	recorder := NewRecorder(r)
 	track := NewTrack(recorder)
-	if match(zngio.NewReader(track, resolver.NewContext())) {
+
+	var zngErr, zeekErr, ndjsonErr, zjsonErr, bzngErr error
```

Comment:
You can use short variable declaration for these in their respective `match` calls.